### PR TITLE
testserver: add option to set root password

### PR DESCRIPTION
**Pull request by tbg**
More than half of the ORMs in Examples-ORMs don't handle secure
connections well out of the box, and some outright don't support
client certs. We want to be able to at least get password auth
working over secure connections, so add an option to set that
up.

### CODE REVIEW

File: testserver/testserver.go [hunk 1]
```diff
@@ -392,6 +407,25 @@ func (ts *testServerImpl) pollListeningURLFile() error {
 	if err != nil {
 		return fmt.Errorf("failed to parse SQL URL: %v", err)
 	}
+	ts.pgURL.orig = *u
+	if pw := ts.serverArgs.rootPW; pw != "" {
+		if pw := ts.serverArgs.rootPW; pw != "" {
```
Comment (rafiss) [inline testserver/testserver.go:412]: Looks like the same if statement twice in a row?

File: testserver/testserver_test.go [hunk 2]
```diff
@@ -114,7 +135,7 @@ func newTenantDBForTest(t *testing.T, secure bool, proxy bool) (*sql.DB, func())
 }
 
 func TestTenant(t *testing.T) {
-	db, stop := newTenantDBForTest(t, false /* secure */, false /* proxy */)
+	db, stop := newTenantDBForTest(t, false /* secure */, false /* proxy */, "")
```
Comment (rafiss) [inline testserver/testserver_test.go:138]: nit: add an inline comment for the empty password

File: testserver/testserver_test.go [hunk 3]
```diff
@@ -32,22 +32,44 @@ func TestRunServer(t *testing.T) {
...
-				return newTenantDBForTest(t, false /* secure */, false /* proxy */)
+				return newTenantDBForTest(t, false /* secure */, false /* proxy */, "")
 			},
 		},
 		{
-			name:          "SecureTenant",
-			instantiation: func(t *testing.T) (*sql.DB, func()) { return newTenantDBForTest(t, true /* secure */, false /* proxy */) },
+			name: "SecureTenant",
+			instantiation: func(t *testing.T) (*sql.DB, func()) {
+				return newTenantDBForTest(t, true /* secure */, false /* proxy */, "")
```
Comment (rafiss) [inline testserver/testserver_test.go:53]: nit: add an inline comment for the empty password

File: testserver/testserver_test.go [hunk 4]
```diff
@@ -32,22 +32,44 @@ func TestRunServer(t *testing.T) {
...
-			instantiation: func(t *testing.T) (*sql.DB, func()) { return newTenantDBForTest(t, true /* secure */, true /* proxy */) },
+			name: "SecureTenantCustomPassword",
+			instantiation: func(t *testing.T) (*sql.DB, func()) {
+				return newTenantDBForTest(t, true /* secure */, false /* proxy */, "AS)*)ASG*")
+			},
+		},
+		{
+			name: "SecureTenantThroughProxy",
+			instantiation: func(t *testing.T) (*sql.DB, func()) {
+				return newTenantDBForTest(t, true /* secure */, true /* proxy */, "")
```
Comment (rafiss) [inline testserver/testserver_test.go:65]: nit: add an inline comment for the empty password

File: testserver/testserver_test.go [hunk 5]
```diff
@@ -32,22 +32,44 @@ func TestRunServer(t *testing.T) {
...
 		},
 		{
-			name:          "Secure",
+			name:          "SecureClientCert",
 			instantiation: func(t *testing.T) (*sql.DB, func()) { return testserver.NewDBForTest(t, testserver.SecureOpt()) },
 		},
+		{
+			name: "SecurePassword",
+			instantiation: func(t *testing.T) (*sql.DB, func()) {
+				return testserver.NewDBForTest(t, testserver.SecureOpt(), testserver.RootPasswordOpt("asd1828585!@$"))
```
Comment (rafiss) [inline testserver/testserver_test.go:41]: not sure if we need different passwords for all these. we could just make a local password variable to reduce the number of magic strings and make the function call self-documenting. no strong preference here, just a thought if you'd like to do that.

**Comment by tbg** [comment]
Ready for another look! Sorry that took so long.

**Review by rafiss — APPROVED** [review]
lgtm!

> Reviewed 1 of 3 files at r1, 2 of 2 files at r2.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach-go/92#-:-MGLt_0mCuOFNtSswSOd:b-7nbn20)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @tbg)

**Comment by tbg** [comment]
TFTR!

**Pull request merged**